crates/spirv-builder/src/lib.rs

```diff
@@ -423,13 +455,8 @@ fn invoke_rustc(builder: &SpirvBuilder) -> Result<PathBuf, SpirvBuilderError> {
     // we do that even in case of an error, to let through any useful messages
     // that ended up on stdout instead of stderr.
     let stdout = String::from_utf8(build.stdout).unwrap();
-    let artifact = get_last_artifact(&stdout);
-
     if build.status.success() {
-        match builder.print_metadata {
-            MetadataPrintout::Full | MetadataPrintout::DependencyOnly => print_deps_of(&artifact),
-            MetadataPrintout::None => (),
-        }
+        let artifact = get_last_artifact(&stdout);
```

Comment:
Please leave this outside the `build.status.success` if statement, due to the comment above it.

Reply:
I found that when If compilation fails, `get_last_artifact` panics, which meant that hot reloading couldn't work. I guess I can make it return an `Option`?

Reply:
Ah, haha, yep, that's unfortunate, rip my bad code :P - yes, please make it an `Option`

Reply:
Handled